Code/Engine/Core/World/Implementation/SpatialSystem_RegularGrid.cpp

```diff
@@ -266,12 +272,71 @@ struct ezSpatialSystem_RegularGrid::Cell
     }
   }
 
+  EZ_FORCE_INLINE void UpdateObject(ezSpatialData* pData)
+  {
+    ezUInt32 mask = pData->m_uiCategoryBitmask;
+    ezUInt32 category = ezMath::FirstBitLow(mask);
+    mask &= mask - 1;
```

Comment:
what is this for ? (the mask &= mask - 1)

Reply:
Ok, I get it now. You want to iterate through all active categories. A comment would be nice, since this logic is not obvious. Potentially even a helper function to put it into.